src/olympia/activity/utils.py

```diff
@@ -0,0 +1,210 @@
+
+        return uuid
+
+    def get_body(self):
+        return self.reply_text
+
+
+def add_email_to_activity_log(email_text):
+    log.debug("Saving from email reply")
+
+    parser = ActivityEmailParser(email_text)
+    if hasattr(parser, 'decode_error'):
+        return False
+
+    uuid = parser.get_uuid()
+
+    if not uuid:
+        return False
+    try:
+        token = ActivityLogToken.objects.get(uuid=uuid)
+    except ActivityLogToken.DoesNotExist:
+        log.error('An email was skipped with non-existing uuid %s.' % uuid)
+        return False
+
+    version = token.version
+    user = token.user
+    if token.is_valid():
+        log_type = None
+
+        if (version.addon.authors.filter(pk=user.pk).exists()):
+            log_type = amo.LOG.DEVELOPER_REPLY_VERSION
+        elif ((version.addon.is_listed and
+               acl.action_allowed_user(user, 'Addons', 'Review')) or
+              (not version.addon.is_listed and
+               acl.action_allowed_user(user, 'Addons', 'ReviewUnlisted'))):
+            log_type = amo.LOG.REVIEWER_REPLY_VERSION
+
+        if log_type:
+            note = log_and_notify(log_type, parser.get_body(), user, version)
+            log.info('A new note has been created (from %s using tokenid %s).'
+                     % (user.id, uuid))
+            token.increment_use()
+            return note
+        else:
+            log.error('%s did not have perms to reply to email thread %s.'
+                      % (user.email, version.id))
+    else:
+        log.error('%s tried to use an invalid activity email token for '
+                  'version %s.' % (user.email, version.id))
+
+    return False
+
+
+def log_and_notify(action, comments, note_creator, version):
+    log_kwargs = {
+        'user': note_creator,
+        'created': datetime.datetime.now(),
+        'details': {
+            'comments': comments,
+            'version': version.version}}
+    note = amo.log(action, version.addon, version, **log_kwargs)
+
+    # Collect reviewers/others involved with this version.
+    log_users = [
+        alog.user for alog in ActivityLog.objects.for_version(version)]
+    # Collect add-on authors (excl. the person who sent the email.)
+    addon_authors = set(version.addon.authors.all()) - {note_creator}
+    # Collect reviewers on the thread (again, excl. the email sender)
+    reviewer_recipients = set(log_users) - addon_authors - {note_creator}
+    author_context_dict = {
+        'name': version.addon.name,
+        'number': version.version,
+        'author': note_creator.name,
+        'comments': comments,
+        'url': version.addon.get_dev_url('versions'),
+        'SITE_URL': settings.SITE_URL,
+    }
+    reviewer_context_dict = author_context_dict.copy()
+    reviewer_context_dict['url'] = absolutify(
+        reverse('editors.review', args=[version.addon.pk], add_prefix=False))
+
+    # Not being localised because we don't know the recipients locale.
```

Comment:
This is still sad, there is https://tools.ietf.org/html/rfc3282 but I've yet to see it being used by some email client :disappointed: